src/client/terminals/codeExecution/terminalReplWatcher.ts

```diff
@@ -0,0 +1,21 @@
+import { Disposable, TerminalShellExecutionStartEvent } from 'vscode';
+import { onDidStartTerminalShellExecution } from '../../common/vscodeApis/windowApis';
+import { sendTelemetryEvent } from '../../telemetry';
+import { EventName } from '../../telemetry/constants';
+
+function checkREPLCommand(command: string): boolean {
+    const lower = command.toLowerCase();
+    // Cover cases for 'py', 'py -3' and 'py -3.x'.
+    const replCommandRegex = /^py(?:\s-3(?:\.\d)?)?$/;
+    return lower.startsWith('python') || lower.startsWith('python3') || replCommandRegex.test(command);
```

Comment:
Starts with `python` should cover starts with `python3`
```suggestion
    return lower.startsWith('python') || replCommandRegex.test(command);
```